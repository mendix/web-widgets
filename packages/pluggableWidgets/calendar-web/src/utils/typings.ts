import { ObjectItem } from "mendix";
import { CalendarProps as ReactCalendarProps } from "react-big-calendar";
import { withDragAndDropProps } from "react-big-calendar/lib/addons/dragAndDrop";

export interface CalendarEvent {
    title: string;
    start: Date;
    end: Date;
    allDay: boolean;
    color?: string;
    item: ObjectItem;
}

export type EventDropOrResize = {
    event: CalendarEvent;
    start: Date;
    end: Date;
};

export interface DragAndDropCalendarProps<TEvent extends object = Event, TResource extends object = object>
    extends ReactCalendarProps<TEvent, TResource>,
        withDragAndDropProps<TEvent, TResource> {}
