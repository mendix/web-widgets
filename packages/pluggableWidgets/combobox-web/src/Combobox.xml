<?xml version="1.0" encoding="utf-8" ?>
<widget id="com.mendix.widget.web.combobox.Combobox" pluginWidget="true" offlineCapable="true" xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../node_modules/mendix/custom_widget.xsd">
    <name>Combo box</name>
    <description />
    <studioProCategory>Input elements</studioProCategory>
    <studioCategory>Display</studioCategory>
    <helpUrl>https://docs.mendix.com/appstore/widgets/combobox</helpUrl>
    <properties>
        <propertyGroup caption="General">
            <!-- OPTIONS SOURCE -->
            <propertyGroup caption="Data source">
                <property key="optionsSourceType" type="enumeration" defaultValue="association" required="true">
                    <caption>Type</caption>
                    <description />
                    <enumerationValues>
                        <enumerationValue key="association">Association</enumerationValue>
                        <enumerationValue key="enumeration">Enumeration</enumerationValue>
                        <enumerationValue key="boolean">Boolean</enumerationValue>
                    </enumerationValues>
                </property>

                <!-- ENUM & BOOLEAN-->
                <property key="attributeEnumeration" type="attribute" required="true">
                    <caption>Attribute</caption>
                    <description />
                    <attributeTypes>
                        <attributeType name="Enum" />
                    </attributeTypes>
                </property>
                <property key="attributeBoolean" type="attribute" required="true">
                    <caption>Attribute</caption>
                    <description />
                    <attributeTypes>
                        <attributeType name="Boolean" />
                    </attributeTypes>
                </property>
                <!-- END ENUM & BOOLEAN -->
                <!-- ASSOCIATION -->
                <property key="attributeAssociation" type="association" selectableObjects="optionsSourceAssociationDataSource" required="true">
                    <caption>Entity</caption>
                    <description />
                    <associationTypes>
                        <associationType name="Reference" />
                        <associationType name="ReferenceSet" />
                    </associationTypes>
                </property>
                <property key="optionsSourceAssociationDataSource" type="datasource" isList="true" required="false">
                    <caption>Selectable objects</caption>
                    <description />
                </property>
                <property key="optionsSourceAssociationCaptionType" type="enumeration" defaultValue="attribute">
                    <caption>Caption type</caption>
                    <description />
                    <enumerationValues>
                        <enumerationValue key="attribute">Attribute</enumerationValue>
                        <enumerationValue key="expression">Expression</enumerationValue>
                    </enumerationValues>
                </property>
                <!-- CAPTIONS -->
                <property key="optionsSourceAssociationCaptionAttribute" type="attribute" dataSource="optionsSourceAssociationDataSource" required="true">
                    <caption>Caption</caption>
                    <description />
                    <attributeTypes>
                        <attributeType name="String" />
                    </attributeTypes>
                </property>
                <property key="optionsSourceAssociationCaptionExpression" type="expression" dataSource="optionsSourceAssociationDataSource" required="true">
                    <caption>Caption</caption>
                    <description />
                    <returnType type="String" />
                </property>
                <!-- END CAPTIONS -->
            </propertyGroup>
            <!-- END OPTIONS SOURCE -->

            <!-- GENERAL -->
            <propertyGroup caption="General">
                <!-- Empty option -->
                <property key="emptyOptionText" type="textTemplate" required="false">
                    <caption>Placeholder text</caption>
                    <description />
                </property>

                <!-- MISC PROPS -->
                <property key="filterType" type="enumeration" defaultValue="contains">
                    <caption>Filter type</caption>
                    <description />
                    <enumerationValues>
                        <enumerationValue key="contains">Contains</enumerationValue>
                        <enumerationValue key="startsWith">Starts-with</enumerationValue>
                        <enumerationValue key="no">No</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="clearable" type="boolean" defaultValue="true">
                    <caption>Clearable</caption>
                    <description />
                </property>
                <!-- END MISC PROPS -->
            </propertyGroup>
            <!-- END GENERAL -->

            <propertyGroup caption="Review">
                <!-- Empty option -->
                <property key="selectionType" type="enumeration" defaultValue="checkbox" required="true">
                    <caption>Selection Type</caption>
                    <description />
                    <enumerationValues>
                        <enumerationValue key="checkbox">Checkboxes</enumerationValue>
                        <enumerationValue key="selectedTag">Selected Item Tags</enumerationValue>
                    </enumerationValues>
                </property>
            </propertyGroup>

            <!-- SYSTEM PROPS -->
            <propertyGroup caption="Label">
                <systemProperty key="Label" />
            </propertyGroup>

            <propertyGroup caption="Conditional visibility">
                <systemProperty key="Visibility" />
            </propertyGroup>
            <propertyGroup caption="Editability">
                <systemProperty key="Editability" />
            </propertyGroup>
            <!-- END SYSTEM PROPS -->
        </propertyGroup>

        <propertyGroup caption="Events">
            <property key="onChangeEvent" type="action" required="false">
                <caption>On change action</caption>
                <description />
            </property>
            <property key="onEnterEvent" type="action" required="false">
                <caption>On enter action</caption>
                <description />
            </property>
            <property key="onLeaveEvent" type="action" required="false">
                <caption>On leave action</caption>
                <description />
            </property>
        </propertyGroup>

        <propertyGroup caption="Accessibility">
            <property key="ariaRequired" type="boolean" defaultValue="false">
                <caption>Aria required</caption>
                <description />
            </property>
        </propertyGroup>
    </properties>
</widget>
