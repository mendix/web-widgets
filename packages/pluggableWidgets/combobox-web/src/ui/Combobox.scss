$cb-brand-primary: #264ae5;
$cb-gray-dark: #606671;
$cb-gray-primary: #ced0d3;
$cb-brand-secondary-light: #e6eaff;
$cb-separator-border: #787d87;
$cb-typography-color: #3b4251;
$cb-box-shadow-color: rgba(0, 0, 0, 0.2);
$cb-selected-item: #e6eaff;
$cb-highlighted-index: #f5f6f6;
$cb-unselected-item: #ffffff;
$cb-hover-color: #ced0d3;
$cb-disabled-color: #f8f8f8;
$cb-disabled-text-color: #6c7180;
$cb-menu-outer-padding: 10px;
.widget-combobox {
    flex-grow: 1;
    position: relative;
    transition: color 150ms ease 0s;
    &-menu {
        position: absolute;
        display: inline;
        border-radius: 7px;
        margin: 4px 0 4px 0;
        width: 100%;
        left: unset;
        padding: $cb-menu-outer-padding 0 0;
        z-index: 1;
        box-shadow: 0px 0px $cb-menu-outer-padding 0px $cb-box-shadow-color;
        background-color: $cb-unselected-item;
        list-style-type: "none";
        &-list {
            padding: 0;
            margin-bottom: 0;
            max-height: 320px;
            overflow-y: auto;
            &:last-child {
                margin-bottom: $cb-menu-outer-padding;
            }
        }
        &-hidden {
            display: none;
        }
        &-header {
            border-bottom: 1px solid $cb-gray-primary;
            &:focus,
            &:focus-within,
            &:hover {
                background-color: $cb-highlighted-index;
            }

            &:has(input[type="checkbox"]:hover, :focus, :focus-within)
                + .widget-combobox-menu-list:not(.widget-combobox-menu-highlighted)
                input[type="checkbox"]:not(:checked):after {
                content: "";
                border-color: $cb-hover-color;
            }
        }

        &-footer {
            border-top: 1px solid $cb-gray-primary;
            padding: $cb-menu-outer-padding;
            &:focus,
            &:focus-within {
                outline: 1px solid $cb-brand-primary;
            }
        }
    }

    &-item {
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        align-content: center;
        align-items: center;
        cursor: pointer;
        user-select: none;
        padding: 6px $cb-menu-outer-padding;
        height: fit-content;
        overflow: hidden;
        color: $cb-typography-color;
        background-color: $cb-unselected-item;

        &-selected {
            background-color: $cb-selected-item;
        }
        &-highlighted,
        &:focus {
            background-color: $cb-highlighted-index;
        }
        & > .widget-combobox-icon-container {
            margin-right: $cb-menu-outer-padding;
        }

        .widget-combobox-caption-text {
            text-overflow: ellipsis;
            overflow: hidden;
            white-space: nowrap;
            flex: 1;
            font-weight: normal;
            margin: 0;
            min-height: 20px;
        }

        &.widget-combobox-no-options {
            justify-content: center;
        }
    }

    .widget-combobox-input-container {
        flex-grow: 1;
        transition: box-shadow 150ms ease 0s;

        &-disabled {
            background-color: $cb-disabled-color;
            pointer-events: none !important;

            &.form-control-static {
                background-color: transparent;
            }
        }
    }

    &-input {
        color: $cb-gray-dark;
        flex-grow: 1;
        border: none;
        padding: 0;

        &:not(:focus) {
            opacity: 0;
        }

        &-nofilter {
            cursor: pointer;
        }
    }

    &-clear-button {
        display: flex;
        align-items: center;
        cursor: pointer;
        background: transparent;
        border: none;
        &-icon {
            transition: color 0.2s ease-in-out;
        }
        path {
            stroke-width: 0;
            transition: stroke-width 0.2s ease-in-out;
        }

        &:focus .widget-combobox-icon-container {
            border-radius: 2px;
            outline: 2px solid $cb-brand-primary;
        }

        &:hover &-icon {
            color: $cb-brand-primary;
            font-weight: bold;

            path {
                stroke-width: 1px;
            }
        }

        & + .widget-combobox-down-arrow {
            border-left: 1px solid $cb-separator-border;
        }
    }
    &-down-arrow {
        display: flex;
        flex-wrap: wrap;
        align-content: center;
        cursor: pointer;
        padding-left: 4px;

        &-icon {
            transition: transform 0.2s;
            &.active {
                transform: rotate(180deg);
                transition: transform 0.2s;
            }
        }
    }

    &-placeholder {
        background-color: $cb-disabled-color !important;
        justify-content: flex-end !important;
        border: 1px solid #e7e7e9;
        cursor: not-allowed;

        &-text {
            color: $cb-typography-color;
            position: absolute;
            left: 0;
            right: 0;
            top: 0;
            bottom: 0;
            text-overflow: ellipsis;
            overflow: hidden;
            white-space: nowrap;
            pointer-events: none;
            display: none;
        }

        &-empty {
            color: $cb-gray-dark;
        }
    }

    &-caption-custom {
        flex: 1;
    }

    &-custom-content {
        position: relative;
        display: flex;
        flex-grow: 1;
        flex-direction: column;

        .widget-combobox-input {
            &:not(:focus),
            &:placeholder-shown {
                position: absolute;
                top: 0;
                left: 0;
                right: 0;
                bottom: 0;
            }
            &:focus:not(:placeholder-shown) + .widget-combobox-placeholder-custom {
                display: none;
            }
        }
    }

    &-selected {
        &-items {
            display: flex;
            flex-grow: 1;
            flex-wrap: wrap;
            position: relative;

            &.widget-combobox-boxes {
                margin: -2px 0;
                .widget-combobox-input {
                    &-nofilter,
                    &:not(:focus) {
                        width: 1px;
                    }
                }
            }

            input:placeholder-shown,
            input:not(:focus) {
                & + .widget-combobox-placeholder-text {
                    display: block;
                }
            }
        }

        &-item {
            color: #000;
            font-size: 12px;
            line-height: 1.334;
            display: inline-flex;
            border-radius: 26px;
            justify-content: center;
            padding: 2px 10px;
            flex-wrap: wrap;
            align-items: center;
            margin: 2px;
            gap: 8px;
            background-color: $cb-brand-secondary-light;
            &-remove-button {
                padding: 0px 0px;
                border: none;
                background-color: transparent;
                cursor: pointer;
                color: $cb-brand-primary;
                span {
                    display: flex;
                    align-items: center;
                }
            }
            &:focus-visible {
                outline: $cb-brand-primary auto 1px;
            }
        }
    }

    &-down-checkbox-icon {
        fill: transparent;
        &:not(.checked) {
            rect {
                stroke: $cb-gray-primary;
            }
            &:hover,
            &:focus {
                path {
                    stroke: $cb-hover-color;
                    stroke-width: 2px;
                    stroke-linecap: round;
                    stroke-linejoin: round;
                }
            }
        }
        &.checked {
            rect {
                fill: $cb-brand-primary;
            }
            path {
                stroke: white;
                stroke-width: 2px;
                stroke-linecap: round;
                stroke-linejoin: round;
            }
        }
    }

    &-icon-container {
        display: flex;
        padding-top: 1px;
    }
}
