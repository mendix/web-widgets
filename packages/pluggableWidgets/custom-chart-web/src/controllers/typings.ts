import { ActionValue, EditableValue } from "mendix";

export interface ControllerProps {
    configurationOptions: string;
    dataAttribute?: EditableValue<string>;
    dataStatic: string;
    eventDataAttribute?: EditableValue<string>;
    layoutAttribute?: EditableValue<string>;
    layoutStatic: string;
    onClick?: ActionValue;
    sampleData: string;
    sampleLayout: string;
}
