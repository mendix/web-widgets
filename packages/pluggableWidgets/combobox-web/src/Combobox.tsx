import { createElement, ReactElement } from "react";
import { useActionEvents } from "./hooks/useActionEvents";
import { ComboboxContainerProps } from "../typings/ComboboxProps";
import { SingleSelection } from "./components/SingleSelection/SingleSelection";
import { MultiSelection } from "./components/MultiSelection/MultiSelection";

import "./ui/Combobox.scss";
import { useGetSelector } from "./hooks/useGetSelector";

export default function Combobox(props: ComboboxContainerProps): ReactElement {
    const actionEvents = useActionEvents(props);
    const selector = useGetSelector(props);

    return (
        <div className="widget-combobox" id={props.id} {...actionEvents}>
            {selector.type === "single" ? (
                <SingleSelection selector={selector} tabIndex={props.tabIndex!} />
            ) : (
                <MultiSelection selector={selector} tabIndex={props.tabIndex!} />
            )}
        </div>
    );
}
