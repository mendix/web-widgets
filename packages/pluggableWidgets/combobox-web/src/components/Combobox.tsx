import classNames from "classnames";
import Downshift from "downshift";
import { createElement, ReactElement, useRef, useState } from "react";
import { useActionEvents } from "../hooks/useActionEvents";
import { ComboboxContainerProps } from "../../typings/ComboboxProps";
import { ClearButton, DownArrow } from "../assets/icons";
import { useDownshiftProps } from "../hooks/useDownshiftProps";
import { useGetSelector } from "../hooks/useGetSelector";
import { ComboboxMenu } from "./ComboboxMenu";
import { Placeholder } from "./Placeholder";

export function Combobox(props: ComboboxContainerProps): ReactElement {
    const inputRef = useRef<HTMLInputElement>(null);
    const comboboxRef = useRef<HTMLDivElement>(null);
    const selector = useGetSelector(props);
    const downshiftProps = useDownshiftProps(
        selector,
        inputRef.current,
        props.emptyOptionText?.value,
        props.onChangeEvent
    );
    const actionEvents = useActionEvents(props);
    const [focused, setFocused] = useState(false);
    const readOnly =
        (props.attributeBoolean?.readOnly || props.attributeEnumeration?.readOnly) ??
        props.attributeAssociation?.readOnly;

    if (selector.status === "unavailable") {
        return <Placeholder />;
    }

    function handleFocus(): void {
        setFocused(true);
    }

    function handleBlur(): void {
        setFocused(false);
    }

    return (
        <Downshift {...downshiftProps}>
            {({
                getInputProps,
                getItemProps,
                selectedItem,
                getMenuProps,
                clearSelection,
                isOpen,
                highlightedIndex,
                getToggleButtonProps
            }): JSX.Element => (
                <div className="widget-combobox" {...actionEvents}>
                    <div
                        ref={comboboxRef}
                        tabIndex={-1}
                        className={classNames("form-control", "widget-combobox-input-container", {
                            "widget-combobox-input-container-active": isOpen || focused,
                            "widget-combobox-input-container-disabled": readOnly
                        })}
                        {...getToggleButtonProps({
                            disabled: readOnly
                        })}
                    >
                        <input
                            id="widget-combobox-input"
                            className="widget-combobox-input"
                            tabIndex={0}
                            ref={inputRef}
                            {...getInputProps({
                                onFocus: handleFocus,
                                onBlur: handleBlur,
                                disabled: readOnly,
                                readOnly: props.filterType === "no"
                            })}
                            placeholder={
                                downshiftProps.selectedItem
                                    ? selector.caption.get(selector.currentValue)
                                    : props.emptyOptionText?.value
                            }
                        />
                        {!readOnly && selector.clearable && selector.currentValue !== null && (
                            <button
                                className="widget-combobox-clear-button"
                                onClick={e => {
                                    e.preventDefault();
                                    e.stopPropagation();
                                    clearSelection(() => selector.setValue(null));
                                }}
                            >
                                <ClearButton />
                            </button>
                        )}
                        <div className="widget-combobox-down-arrow">
                            <DownArrow />
                        </div>
                    </div>
                    <ComboboxMenu
                        comboboxSize={comboboxRef.current?.getBoundingClientRect()}
                        selector={selector}
                        filterType={props.filterType}
                        selectedItem={selectedItem}
                        getMenuProps={getMenuProps}
                        getItemProps={getItemProps}
                        isOpen={isOpen}
                        highlightedIndex={highlightedIndex}
                    />
                </div>
            )}
        </Downshift>
    );
}
