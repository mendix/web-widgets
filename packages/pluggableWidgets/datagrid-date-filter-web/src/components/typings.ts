import { Date_InputFilterInterface } from "@mendix/widget-plugin-filtering/typings/InputFilterInterface";

export interface DateFilterProps {
    filterStore: Date_InputFilterInterface;
    parentChannelName?: string;
}
