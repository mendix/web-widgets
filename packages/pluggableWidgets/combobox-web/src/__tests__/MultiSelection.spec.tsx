import {
    buildListExpression,
    dynamicValue,
    EditableValueBuilder,
    ListAttributeValueBuilder,
    ListValueBuilder,
    ReferenceSetValueBuilder
} from "@mendix/pluggable-test-utils";
import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import { ObjectItem, DynamicValue } from "mendix";
import { createElement } from "react";
import { ComboboxContainerProps } from "../../typings/ComboboxProps";
import Combobox from "../Combobox";

describe("Combo box (Association)", () => {
    let defaultProps: ComboboxContainerProps;
    beforeEach(() => {
        defaultProps = {
            name: "comboBox",
            id: "comboBox1",
            optionsSourceType: "association",
            attributeAssociation: new ReferenceSetValueBuilder().withValue([{ id: "111" }] as ObjectItem[]).build(),
            attributeEnumeration: new EditableValueBuilder<string>().build(),
            attributeBoolean: new EditableValueBuilder<boolean>().build(),
            optionsSourceAssociationDataSource: ListValueBuilder().withItems([
                { id: "111" },
                { id: "222" },
                { id: "333" },
                { id: "444" }
            ] as ObjectItem[]),
            optionsSourceAssociationCaptionType: "expression",
            optionsSourceAssociationCaptionAttribute: new ListAttributeValueBuilder<string>().build(),
            optionsSourceAssociationCaptionExpression: buildListExpression("$currentObject/CountryName"),
            emptyOptionText: dynamicValue("Select an option 111"),
            ariaRequired: true,
            clearable: true,
            filterType: "contains",
            selectionType: "checkbox"
        };
        if (defaultProps.optionsSourceAssociationCaptionType === "expression") {
            defaultProps.optionsSourceAssociationCaptionExpression!.get = i => {
                return {
                    value: `${i.id}`,
                    status: "available"
                } as DynamicValue<string>;
            };
        }
    });
    it("renders combobox widget", () => {
        const component = render(<Combobox {...defaultProps} />);
        expect(component.container).toMatchSnapshot();
    });
    it("renders placeholder component in case of unavailable status", () => {
        defaultProps.attributeAssociation = new ReferenceSetValueBuilder().isUnavailable().build();
        const { container } = render(<Combobox {...defaultProps} />);
        expect(container.getElementsByClassName("widget-combobox-placeholder")).toHaveLength(1);
    });
    it("toggles combobox menu on: input FOCUS / BLUR", async () => {
        const component = render(<Combobox {...defaultProps} />);
        const toggleButton = await component.findByRole("combobox");
        fireEvent.focus(toggleButton);
        expect(component.getAllByRole("option")).toHaveLength(4);
        fireEvent.blur(toggleButton);
        expect(component.queryAllByRole("option")).toHaveLength(0);
    });
    it("toggles combobox menu on: input TOGGLE BUTTON", async () => {
        const component = render(<Combobox {...defaultProps} />);
        const toggleButton = await component.container.querySelector(".widget-combobox-down-arrow")!;
        fireEvent.click(toggleButton);
        expect(component.getAllByRole("option")).toHaveLength(4);
        fireEvent.click(toggleButton);
        expect(component.queryAllByRole("option")).toHaveLength(0);
    });
    it("adds new item to inital selected item", async () => {
        const component = render(<Combobox {...defaultProps} />);
        const input = (await component.findByRole("combobox")) as HTMLInputElement;
        fireEvent.focus(input);
        const option1 = await component.findByText("222");
        fireEvent.click(option1);
        expect(component.container).toMatchSnapshot();
        expect(defaultProps.attributeAssociation?.setValue).toBeCalled();
        expect(component.queryAllByRole("option")).toHaveLength(4);
        expect(defaultProps.attributeAssociation?.value).toEqual([{ id: "111" }, { id: "222" }]);
    });
    it("removes selected item", async () => {
        const component = render(<Combobox {...defaultProps} />);

        const input = (await component.findByRole("combobox")) as HTMLInputElement;
        fireEvent.focus(input);

        const option1 = await component.findByText("222");
        fireEvent.click(option1);
        expect(defaultProps.attributeAssociation?.setValue).toBeCalled();
        expect(component.queryAllByRole("option")).toHaveLength(4);
        expect(defaultProps.attributeAssociation?.value).toEqual([{ id: "111" }, { id: "222" }]);

        const clearButton = await component.container.getElementsByClassName("widget-combobox-clear-button")[0];
        fireEvent.click(clearButton);
        expect(defaultProps.attributeAssociation?.value).toEqual([]);
    });
});
