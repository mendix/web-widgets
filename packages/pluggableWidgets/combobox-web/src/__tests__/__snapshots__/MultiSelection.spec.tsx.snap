// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Combo box (Association) renders combobox widget 1`] = `
<div>
  <div
    class="widget-combobox"
  >
    <div
      class="widget-combobox-input-container form-control"
      id="downshift-0-toggle-button"
      tabindex="-1"
    >
      <div
        class="widget-combobox-selected-items widget-combobox-text"
      >
        <input
          aria-activedescendant=""
          aria-autocomplete="list"
          aria-controls="downshift-0-menu"
          aria-expanded="false"
          aria-labelledby="comboBox1-label"
          autocomplete="off"
          class="widget-combobox-input"
          id="comboBox1"
          placeholder=" "
          role="combobox"
          value=""
        />
        <div
          class="widget-combobox-placeholder-text"
        >
          111
        </div>
      </div>
      <button
        aria-label="Clear selection"
        class="widget-combobox-clear-button"
      >
        <span
          class="widget-combobox-icon-container"
        >
          <svg
            class="widget-combobox-clear-button-icon"
            height="14"
            viewBox="0 0 32 32"
            width="14"
          >
            <path
              d="M27.71 5.71004L26.29 4.29004L16 14.59L5.71004 4.29004L4.29004 5.71004L14.59 16L4.29004 26.29L5.71004 27.71L16 17.41L26.29 27.71L27.71 26.29L17.41 16L27.71 5.71004Z"
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
      </button>
      <div
        class="widget-combobox-down-arrow"
      >
        <span
          class="widget-combobox-icon-container"
        >
          <svg
            class="widget-combobox-down-arrow-icon mx-icon-lined mx-icon-chevron-down"
            height="16"
            viewBox="0 0 32 32"
            width="16"
          >
            <path
              d="M16 23.41L4.29004 11.71L5.71004 10.29L16 20.59L26.29 10.29L27.71 11.71L16 23.41Z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="widget-combobox-menu widget-combobox-menu-hidden"
      style="visibility: hidden; position: fixed;"
    >
      <div
        class="widget-combobox-menu-header widget-combobox-item"
        tabindex="0"
      >
        <span
          class="widget-combobox-menu-header-select-all-button widget-combobox-icon-container"
        >
          <input
            checked=""
            class="three-state-checkbox"
            id="comboBox1-select-all-button"
            type="checkbox"
          />
        </span>
        <label
          class="widget-combobox-caption-text"
          for="comboBox1-select-all-button"
        >
          Select All
        </label>
      </div>
      <ul
        aria-labelledby="comboBox1-label"
        class="widget-combobox-menu-list"
        id="downshift-0-menu"
        role="listbox"
      />
    </div>
  </div>
</div>
`;

exports[`Combo box (Association) toggles combobox menu on: input CLICK(focus) / BLUR 1`] = `
<div>
  <div
    class="widget-combobox"
  >
    <div
      class="widget-combobox-input-container form-control"
      id="downshift-2-toggle-button"
      tabindex="-1"
    >
      <div
        class="widget-combobox-selected-items widget-combobox-text"
      >
        <input
          aria-activedescendant=""
          aria-autocomplete="list"
          aria-controls="downshift-2-menu"
          aria-expanded="false"
          aria-labelledby="comboBox1-label"
          autocomplete="off"
          class="widget-combobox-input"
          id="comboBox1"
          placeholder=" "
          role="combobox"
          value=""
        />
        <div
          class="widget-combobox-placeholder-text"
        >
          111
        </div>
      </div>
      <button
        aria-label="Clear selection"
        class="widget-combobox-clear-button"
      >
        <span
          class="widget-combobox-icon-container"
        >
          <svg
            class="widget-combobox-clear-button-icon"
            height="14"
            viewBox="0 0 32 32"
            width="14"
          >
            <path
              d="M27.71 5.71004L26.29 4.29004L16 14.59L5.71004 4.29004L4.29004 5.71004L14.59 16L4.29004 26.29L5.71004 27.71L16 17.41L26.29 27.71L27.71 26.29L17.41 16L27.71 5.71004Z"
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
      </button>
      <div
        class="widget-combobox-down-arrow"
      >
        <span
          class="widget-combobox-icon-container"
        >
          <svg
            class="widget-combobox-down-arrow-icon mx-icon-lined mx-icon-chevron-down"
            height="16"
            viewBox="0 0 32 32"
            width="16"
          >
            <path
              d="M16 23.41L4.29004 11.71L5.71004 10.29L16 20.59L26.29 10.29L27.71 11.71L16 23.41Z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="widget-combobox-menu widget-combobox-menu-hidden"
      style="visibility: visible; position: fixed; width: 0px; top: 0px; left: 0px;"
    >
      <div
        class="widget-combobox-menu-header widget-combobox-item"
        tabindex="0"
      >
        <span
          class="widget-combobox-menu-header-select-all-button widget-combobox-icon-container"
        >
          <input
            checked=""
            class="three-state-checkbox"
            id="comboBox1-select-all-button"
            type="checkbox"
          />
        </span>
        <label
          class="widget-combobox-caption-text"
          for="comboBox1-select-all-button"
        >
          Select All
        </label>
      </div>
      <ul
        aria-labelledby="comboBox1-label"
        class="widget-combobox-menu-list"
        id="downshift-2-menu"
        role="listbox"
      />
    </div>
  </div>
</div>
`;

exports[`Combo box (Association) toggles combobox menu on: input TOGGLE BUTTON 1`] = `
<div>
  <div
    class="widget-combobox"
  >
    <div
      class="widget-combobox-input-container form-control"
      id="downshift-7-toggle-button"
      tabindex="-1"
    >
      <div
        class="widget-combobox-selected-items widget-combobox-text"
      >
        <input
          aria-activedescendant=""
          aria-autocomplete="list"
          aria-controls="downshift-7-menu"
          aria-expanded="false"
          aria-labelledby="comboBox1-label"
          autocomplete="off"
          class="widget-combobox-input"
          id="comboBox1"
          placeholder=" "
          role="combobox"
          value=""
        />
        <div
          class="widget-combobox-placeholder-text"
        >
          111
        </div>
      </div>
      <button
        aria-label="Clear selection"
        class="widget-combobox-clear-button"
      >
        <span
          class="widget-combobox-icon-container"
        >
          <svg
            class="widget-combobox-clear-button-icon"
            height="14"
            viewBox="0 0 32 32"
            width="14"
          >
            <path
              d="M27.71 5.71004L26.29 4.29004L16 14.59L5.71004 4.29004L4.29004 5.71004L14.59 16L4.29004 26.29L5.71004 27.71L16 17.41L26.29 27.71L27.71 26.29L17.41 16L27.71 5.71004Z"
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
      </button>
      <div
        class="widget-combobox-down-arrow"
      >
        <span
          class="widget-combobox-icon-container"
        >
          <svg
            class="widget-combobox-down-arrow-icon mx-icon-lined mx-icon-chevron-down"
            height="16"
            viewBox="0 0 32 32"
            width="16"
          >
            <path
              d="M16 23.41L4.29004 11.71L5.71004 10.29L16 20.59L26.29 10.29L27.71 11.71L16 23.41Z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="widget-combobox-menu widget-combobox-menu-hidden"
      style="visibility: hidden; position: fixed;"
    >
      <div
        class="widget-combobox-menu-header widget-combobox-item"
        tabindex="0"
      >
        <span
          class="widget-combobox-menu-header-select-all-button widget-combobox-icon-container"
        >
          <input
            checked=""
            class="three-state-checkbox"
            id="comboBox1-select-all-button"
            type="checkbox"
          />
        </span>
        <label
          class="widget-combobox-caption-text"
          for="comboBox1-select-all-button"
        >
          Select All
        </label>
      </div>
      <ul
        aria-labelledby="comboBox1-label"
        class="widget-combobox-menu-list"
        id="downshift-7-menu"
        role="listbox"
      />
    </div>
  </div>
</div>
`;
